CCP_App/src/main/java/com/example/demo2/service/UserService.java

```diff
@@ -55,33 +60,35 @@ public List<User> getAllUsers() {
 	 * @param user User firstName and lastName
 	 */
 	public void insertUser(UserViewModel user) {
-			User userData = new User(user.getFirstName(), user.getLastName(), user.getUserName(), user.getEmail());
-			userRepository.insert(userData);
+		User userData = userGateway.convertUserToUserModel(user);
+		userRepository.insert(userData);
 	}
 
 	/**
 	 * Updating user details and checking that the username field has not been updated .
 	 *
 	 * @param user User firstName , lastName , email and username.
 	 */
-	public void updateUser(UserViewModel user) {
+	public void updateUser(UserViewModel model)
+			throws NoElementFoundException, NonUpdateFieldUserTableException, DatabaseNotAccessibleException {
 
-		User foundUser;
+		try {
 
-		var data = userRepository.findById(user.getId());
-		if (!data.isEmpty()) {
-			foundUser = data.get();
-			if (user.getUserName().equals(foundUser.getUserName())) {
-				foundUser.setFirstName(user.getFirstName());
-				foundUser.setLastName(user.getLastName());
-				foundUser.setEmail(user.getEmail());
-				userRepository.save(foundUser);
+			var data = userRepository.findById(model.getId());
+			if (!data.isEmpty()) {
+				User user = userGateway.convertUserToUserModel(model);
+				User foundUser = data.get();
+				if (user.getUserName().equals(foundUser.getUserName())) {
+					userRepository.save(user);
 
+				} else {
+					throw new NonUpdateFieldUserTableException();
+				}
 			} else {
-				throw new NonUpdateFieldException();
+				throw new NoElementFoundException();
 			}
-		} else {
-			throw new NoElementFountException();
+		} catch (DatabaseNotAccessibleException exception) {
```

Comment:
very BOLD assumption on naming it DatabaseNotAccessibleException . Using a try catch like this there is no guarantee that it was a database not accessable exception or a logic or db usage exception.

Also I dont think that this exception can ever be hit. Try testing it by starting your application and then using docker to completely stop your mongodb and see if this works